### PR TITLE
LibJS: Add bytecode support for regexp literals

### PR DESCRIPTION
**Pull request by mattco98**

### CODE REVIEW

File: Userland/Libraries/LibJS/Bytecode/ASTCodegen.cpp [hunk 1]
```diff
@@ -258,6 +259,39 @@ void StringLiteral::generate_bytecode(Bytecode::Generator& generator) const
...
+            flags = flags | static_cast<u16>(ECMAScriptFlags::Multiline);
+            break;
+        case 's':
+            flags = flags | static_cast<u16>(ECMAScriptFlags::SingleLine);
+            break;
+        case 'u':
+            flags = flags | static_cast<u16>(ECMAScriptFlags::Unicode);
+            break;
+        case 'y':
+            flags = flags | static_cast<u16>(ECMAScriptFlags::Sticky);
```
Comment (linusg) [inline Userland/Libraries/LibJS/Bytecode/ASTCodegen.cpp:285]: `flags |= ...`

**Review by linusg — CHANGES_REQUESTED** [review]
What's the point of turning the AST node's `m_flags` string into `ECMAScriptFlags`, only to then turn them back into a string for `RegExpObject::create()`, which will then parse them again?!

File: Userland/Libraries/LibJS/Bytecode/Op.h [hunk 2]
```diff
@@ -162,6 +162,24 @@ class NewObject final : public Instruction {
     void replace_references_impl(BasicBlock const&, BasicBlock const&) { }
 };
 
+class NewRegExpLiteral final : public Instruction {
```
Comment (awesomekling) [inline Userland/Libraries/LibJS/Bytecode/Op.h:165]: `Literal` feels like a parsing-stage concept, I think we can just call this `NewRegExp` :)

File: Userland/Libraries/LibJS/Bytecode/ASTCodegen.cpp (hunk 1 above)
Comment (alimpfard) [inline Userland/Libraries/LibJS/Bytecode/ASTCodegen.cpp:285]: PSA: You can now use `to_underlying` to case enum classes to their underlying type instead.

File: Userland/Libraries/LibJS/Bytecode/Op.cpp [hunk 3]
```diff
@@ -169,6 +170,28 @@ void NewObject::execute_impl(Bytecode::Interpreter& interpreter) const
...
+
+    if (m_flags & static_cast<u16>(ECMAScriptFlags::Global))
+        flags_builder.append('g');
+    if (m_flags & static_cast<u16>(ECMAScriptFlags::Insensitive))
+        flags_builder.append('i');
+    if (m_flags & static_cast<u16>(ECMAScriptFlags::Multiline))
+        flags_builder.append('m');
+    if (m_flags & static_cast<u16>(ECMAScriptFlags::SingleLine))
+        flags_builder.append('s');
+    if (m_flags & static_cast<u16>(ECMAScriptFlags::Unicode))
```
Comment (alimpfard) [inline Userland/Libraries/LibJS/Bytecode/Op.cpp:187]: Same `to_underlying()` comment here.

**Pull request merged**